xclim/data/fr.json

```diff
@@ -553,6 +553,18 @@
     "comment": "Formule originale prise de Jackson, D. I., & Cherry, N. J. (1988)",
     "abstract": "Indice climatique basé sur la température moyenne du mois le plus chaud et un coefficient basé sur la latitude pour tenir compte de la durée plus longue du jour favorisant les conditions de croissance. Développé spécifiquement pour la viticulture. Température moyenne du mois le plus chaud multipliée par la différence de coéffient de facteur latitude moins la latitude."
   },
+  "AUSTRALIAN_HARDINESS_ZONES": {
+    "long_name": "Échelle des températures minimales annuelles moyennes pendant une période donnée (Australian National Botanic Gardens)",
+    "description": "Indice climatique basé sur une moyenne mobile de la température annuelle minimale de {window} années. Développé spécifiquement pour aider à déterminer l'adéquation des plantes aux régions géographiques. Le système de classification des jardins botaniques nationaux australiens divise les catégories en zones de 5 degrés Celsius, allant de moins de -15 degrés Celsius à plus de 20 degrés Celsius.",
```

Comment:
```suggestion
    "description": "Indice climatique basé sur une moyenne mobile de {window} années du minimum annuel de la température minimale. Développé spécifiquement pour aider à déterminer l'adéquation des plantes aux régions géographiques. Le système de classification des jardins botaniques nationaux australiens divise les catégories en zones de 5 degrés Celsius, allant de moins de -15 degrés Celsius à plus de 20 degrés Celsius.",
```

Reply:
À valider

Reply:
https://planthardiness.ars.usda.gov/: "The map is based on the average annual minimum winter temperature, divided into 10-degree F zones and further divided into 5-degree F half-zones"

Reply:
From the website:

> "The statistic used by the USDA is the average annual minimum temperature. This causes immediate confusion. In Australia we use this term to mean the average minimum temperature over the whole year, whereas the USDA means the average, over ten or more years, of the very lowest temperature observed for each year for each meteorological station. I prefer to call the USDA term the average annual lowest temperature."

I suppose this suggests that the averaging window should be `1` for anbg`?

Reply:
Perhaps we could address this in the wrapped indicator with:
```python
    parameters={
        "method": "anbg"
        "window": {"default": 1}
    }
```

Reply:
Therefore, we would be allowing a bit more general behaviour with the indice, since `window` could be changed even though we choose "usda" or "anbg". So, we allow small variations of the definition of the indices only though the use of the index. I think it could make sense indeed to allow more general behaviour with the index name `hardiness-zones`, which is less specific than the indicators : `usda-hardiness-zones` and so on. 

However, that is a weird control on the behaviour though, because we can only change the window, and moreover this is allowed to be changed only when a specific method is chosen (which in principle, already has a defined window). Maybe we should put the zone definitions (`zone_min`, `zone_max`, `step_size`) in the index signature, and allow to either call with "method==usda" or specify (`zone_min`, `zone_max`, `step_size`, `window`)? Another idea is to simply remove `method` from the index and control this exclusively through the indicator (by choosing appropriate  (`zone_min`, `zone_max`, `step_size`, `window`) ) , although I'm not sure this is similar to other comparable indicators in xclim, I'd have to compare with other cases

Reply:
If I may, I suspect you are going too deep for this. Not that the generalization will never be useful, but... We only have two "zone indices" examples, this is not much to plan a generalization. You might be able to think out a plan and get something that looks flexible, but what if the next similar index does not fit ? I would suggest sticking to something simple and a bit specific, waiting for new similar indices to come along before generalizing.

Reply:
I think the complication here is that hardiness zones can become very specialized. That said, the binning logic proposed here is useful for all of them.

One would think that the Canadian Hardiness Zones would the next logical step, but it appears to be quite complex ([Wikipedia](https://en.wikipedia.org/wiki/Hardiness_zone#Canadian_hardiness_zones); [source article](https://cdnsciencepub.com/doi/10.4141/cjps67-064))

The Wikipedia page for hardiness zones also makes mention of the American Horticultural Society's [Heat Zones](https://ahsgardening.org/about-us/news-press/cool_timeline/heat-zone-map-developed/). This metric uses the average number of days exceeding a given threshold, much less complicated, but unfortunately, it seems this was never published academically, so I'm not sure how much effort should be spent including it.

In both cases, the zoning logic is something that could be very useful. In fact, many of the agricultural indices (Winkler and Huglin indices, Cool Night index, Dryness Index, others) could benefit from the averaging and zoning logic. I'm tempted to build on top of them with this new generic index, honestly. 

I think we should start somewhere, though. As far as I'm concerned, this implementation works as expected and is not overly-complicated. The indicator is clear enough for general users as well. Other thoughts?

Reply:
> From the website:
> 
> > "The statistic used by the USDA is the average annual minimum temperature. This causes immediate confusion. In Australia we use this term to mean the average minimum temperature over the whole year, whereas the USDA means the average, over ten or more years, of the very lowest temperature observed for each year for each meteorological station. I prefer to call the USDA term the average annual lowest temperature."
> 
> I suppose this suggests that the averaging window should be `1` for anbg`?

Further down:

> "I have used the same statistic for Australia, but rather than use US zones directly I have modified the limits for each zone."

@huard 

Despite the confusing way this was written, I believe that the ANBG method is meant to be the exact same as that of USDA (mean of annual tn_min over years), averaging window included. I'll fix this up in the code and resolve this!

Reply:
I was mostly worried about the way the algorithm was translated into words (not the other way around), and made the change suggestion only for one both assuming they would apply to both.